modules/nextflow/src/main/groovy/nextflow/executor/AbstractGridExecutor.groovy

```diff
@@ -36,7 +36,7 @@ import org.apache.commons.lang.StringUtils
  */
 @Slf4j
 @CompileStatic
-abstract class AbstractGridExecutor extends Executor {
+abstract class AbstractGridExecutor extends Executor implements TaskArrayAware {
```

Comment:
There's no real reason to `AbstractGridExecutor` implementing `TaskArrayAware`. Better to move it to the concrete classes really supporting it